server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java

```diff
@@ -173,148 +176,170 @@ public String toString() {
         }
     }
 
-    private void trustForBlobId(String blobId, String username) {
-        String serializedAttachmentAccessToken = 
-            with()
-                .header("Authorization", userStepdefs.tokenByUser.get(username).serialize())
-            .post("/download/" + blobId)
-            .then()
-                .extract()
-                .asString();
+    private void trustForBlobId(String blobId, String username) throws ClientProtocolException, IOException, URISyntaxException {
+        Response tokenGenerationResponse = Request.Post(mainStepdefs.baseUri().setPath("/download/" + blobId).build())
+            .addHeader("Authorization", userStepdefs.tokenByUser.get(username).serialize())
+            .execute();
+        String serializedAttachmentAccessToken = tokenGenerationResponse.returnContent().asString();
         attachmentAccessTokens.put(
                 new AttachmentAccessTokenKey(username, blobId),
                 AttachmentAccessToken.from(
                     serializedAttachmentAccessToken,
                     blobId));
     }
 
-    @When("^\"([^\"]*)\" downloads \"([^\"]*)\" with a valid authentication token$")
+    @When("^\"([^\"]*)\" downloads \"([^\"]*)\" with a valid authentication token but a bad blobId$")
     public void downloadsWithValidToken(String username, String attachmentId) throws Throwable {
```

Comment:
because we ask for an access token on the good blobId then use a different blobId on following query